projects/packages/forms/src/modules/file-field/view.js

```diff
@@ -51,116 +113,166 @@ const addFileToContext = file => {
 		error = config.i18n.invalidType;
 	}
 
+	// Get all files that don't have an error properly
+	const validFiles = context.files.filter( fileInfo => ! fileInfo.error );
+
 	// Check if the user is trying to add more files then allowed.
-	if ( context.maxFiles < context.files.length + 1 ) {
+	if ( context.maxFiles < validFiles.length + 1 ) {
 		error = config.i18n.maxFiles;
 	}
 
-	const fileId = performance.now() + '-' + Math.random();
+	const clientFileId = performance.now() + '-' + Math.random();
 
 	// Update the context.
 	context.hasFiles = true;
+
 	context.files.push( {
 		name: file.name,
 		formattedSize: formatBytes( file.size, 2 ),
-		hasToken: false,
+		isUploaded: false,
 		hasError: !! error,
-		id: fileId,
+		id: clientFileId,
 		error,
 	} );
 
+	const uploadFileWithScope = withScope( uploadFile.bind( this, file, clientFileId ) );
```

Comment:
The scope is automatically transmitted in synchronous functions and async actions (the ones using generators), so instead of using `withScope` you could move this to `actions` and just call `actions.uploadFile`.